Sources/StreamChat/Query/Filter+ChatChannel.swift

```diff
@@ -68,28 +68,28 @@ extension Filter where Scope == ChannelListFilterScope {
     ) -> NSPredicate? {
         switch op {
         case .and where mappedValue is [Filter<Scope>]:
-            guard let filters = mappedValue as? [Filter<Scope>] else {
+            guard let predicates = (mappedValue as? [Filter<Scope>])?.compactMap(\.predicate), predicates.count > 0 else {
```

Comment:
@kaijietti Thank you for the PR. I have verified this change. 
 
Could you please replace `predicates.count > 0` with `!predicates.isEmpty` due to our formatting rules in our repository and I will then proceed with merging this PR after verifying that it passes all the CI checks (I will run it separately).

Reply:
@laevandus I replaced in the latest commit, please take a look, thanks!

Reply:
@kaijietti @laevandus Would be nice to add this edge case to the `ChatChannelListController_Tests.test_filterPredicate_XXX...` tests. So that we do not have regressions in th future 🙏

Reply:
Note, I ran these changes on a CI in a separate [job](https://github.com/GetStream/stream-chat-swift/actions/runs/11779894449) and all was green.